packages/codegen-ui-react/lib/__tests__/imports/import-collection.test.ts

```diff
@@ -105,6 +106,88 @@ describe('ImportCollection', () => {
+      importCollection.addGraphqlMutationImport('createPost');
+      assertImportCollectionMatchesSnapshot(importCollection);
+    });
+  });
+
+  describe('addGraphqlQueryImport', () => {
+    it('should import query correctly', () => {
+      const importCollection = new ImportCollection({ rendererConfig: rendererConfigWithGraphQL });
+      importCollection.addGraphqlQueryImport('listPosts');
+      assertImportCollectionMatchesSnapshot(importCollection);
+    });
+  });
+
+  describe('addGraphqlSubscriptionImport', () => {
+    it('should import subscription correctly', () => {
+      const importCollection = new ImportCollection({ rendererConfig: rendererConfigWithGraphQL });
+      importCollection.addGraphqlSubscriptionImport('onCreatePost');
+      assertImportCollectionMatchesSnapshot(importCollection);
+    });
+  });
+
+  describe('addModelImport', () => {
+    it('should use path from rendererConfig if using GraphQL', () => {
+      const importCollection = new ImportCollection({ rendererConfig: rendererConfigWithGraphQL });
+      importCollection.addModelImport('Post');
+      assertImportCollectionMatchesSnapshot(importCollection);
+    });
+
+    it('should use local models constant if using DataStore', () => {
+      const importCollection = new ImportCollection();
+      importCollection.addModelImport('Post');
+      assertImportCollectionMatchesSnapshot(importCollection);
+    });
+  });
+
+  describe('getMappedModelAlias', () => {
+    it('should successfully retrieve model alias when using DataStore', () => {
+      const importCollection = new ImportCollection();
+      importCollection.ingestComponentMetadata({
+        componentNameToTypeMap: { TextInput: 'TextField' },
+        hasAuthBindings: false,
+        requiredDataModels: [],
+        stateReferences: [],
+      });
+      importCollection.addModelImport('TextField0');
+      importCollection.addModelImport('TextField');
+      importCollection.addModelImport('TestModel');
+      importCollection.addModelImport('ButtonProps');
+      importCollection.addModelImport('React');
+      expect(importCollection.getMappedModelAlias('TextField')).toEqual('TextField1');
+      expect(importCollection.getMappedModelAlias('TestModel')).toEqual('TestModel');
+      expect(importCollection.getMappedModelAlias('TextField0')).toEqual('TextField0');
+      expect(importCollection.getMappedModelAlias('ButtonProps')).toEqual('ButtonProps0');
+      expect(importCollection.getMappedModelAlias('React')).toEqual('React0');
+    });
+
+    // TODO: import collection currently does not alias import collisions
+    //   unless it's explicitly for the ImportSource.LOCAL_MODELS package
+    it('should successfully retrieve model alias when using GraphQL', () => {
+      const importCollection = new ImportCollection({ rendererConfig: rendererConfigWithGraphQL });
+      // importCollection.ingestComponentMetadata({
```

Comment:
Can we clean up all the commented out pieces?

Reply:
I wanted to keep this as-is since the intention is to be able to fully uncomment this test once import collection has been updated to address the TODO above.